aip/client-libraries/4222.md

```diff
@@ -154,6 +154,41 @@ If none of the routing parameters matched their respective fields, the routing h
 Much like URL parameters, if there is more than one key-value pair to be sent, the `&`
 character is used as the separator.
 
+### `path_template` syntax
+
+As seen in the above examples, the `path_template` can use a variety of symbols
+that are interpreted by code generators during conversion to regular expressions
+or non-regular expression matcher implementations. The syntax for these symbols
+is as follows:
+
+- Segment delimiters:
+  `path_template`:
+  - `/`: The traditional delimiter for resource names and HTTP paths.
+  - `:`: Often, but not exclusively, used to separate a path segment from a
+    custom method name segment.
+- Wildcards:
+  - `*`: Corresponds to 1 or more non-`/` symbols. The regex describing it is
+    `[^\/]+`.
+  - `**`: Corresponds to 0 or more symbols or segments (so it can be empty).
+    Consumes segment delimiters and does not match unless the delimiter has
+    trailing content. The regular expression is `([:\/].*)?`.
+  - All wildcards **must** be preceded by an acceptable delimiter.
```

Comment:
I think wildcards should only be used _in between_ delimiters except for the last segment, which could have _only_ a preceding delimiter

Reply:
Ok, i think that is a good distinction to make. Let me ask this though: Do you think folks might want to do something like `projects/foo/locations/us-*` to, say, send US locations via a different key than EMEA locations? Just spitballing

Reply:
Ok so I've updated it to say that `*` must only be used _between_ segment delimiters, and `**` must only be used as following the final segment delimiter e.g. `.../**`/ WDYT?

Reply:
That's great and it covers pretty much everything! I _think_ there is only one special case that `{projects=**}` is also accepted, not sure we should add it here  or add it as a special case in the section for `**`.

Reply:
Ok I changed `**` to say it "must only appear as the final segment" (which `{key=**}` conforms to, technically) and I've updated the named segment section to say "one *should* use the simpler `{key}` syntax" instead of explicitly configuring `{key=**}`.